src/backend/columnar/columnar_metadata.c

```diff
@@ -1294,10 +1306,63 @@ ColumnarStorageUpdateIfNeeded(Relation rel, bool isUpgrade)
 	GetHighestUsedAddressAndId(storageId, &highestOffset, &highestId);
 
 	uint64 reservedStripeId = highestId + 1;
-
-	/* XXX: should be set properly */
-	uint64 reservedRowNumber = 0;
+	uint64 reservedRowNumber = ColumnarStorageRowCount(storageId);
 	uint64 reservedOffset = highestOffset + 1;
 	ColumnarStorageUpdateCurrent(rel, isUpgrade, reservedStripeId,
 								 reservedRowNumber, reservedOffset);
+
+	ColumnarStripeSetFirstRowNumberForStorage(storageId);
```

Comment:
Does it make sense to populate `stripe.first_row_number` within `ColumnarStorageUpdateIfNeeded`, or should I write another udf for metadata update, or update the comments for `ColumnarStorageUpdateIfNeeded` ?

Reply:
we don't do that here anymore, now we populate `first_row_number` upgrade script